src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Debug/VBDefineConstantsEncoding.cs

```diff
@@ -18,8 +18,20 @@ public IEnumerable<(string Name, string Value)> Parse(string value)
             }
             foreach (var entry in ReadEntries(value))
             {
-                var (entryKey, entryValue) = SplitEntry(entry);
-                yield return (DecodeCharacters(entryKey), StripQuotes(DecodeCharacters(entryValue)));
+                (string, string) resultingEntry = ("", "");
+                try
+                {
+                    (string entryKey, string entryValue) = SplitPairedEntry(entry);
+                    resultingEntry = (DecodeCharacters(entryKey), DecodeCharacters(entryValue));
+                }
+                //if not key value pair entry, tries to parse for single value entry
+                //empty string in the section position signifies single value entry
```

Comment:
I wonder if we can use an analyzer to suggest this in the IDE rather than code review.

Reply:
Not sure if this is a convention but my rule-of-thumb has been:
- Commented text uses `//` with a space after it
- Commented code uses only `//`

There are some situations where it makes sense to check in commented code. But this is normally what I do when developing. Pressing `Ctrl+K+C` in VS doesn't add that space for code, so that's why I do the differentiation. I think of one as a "human made" comment (with the space) and the other as an "IDE" comment.

Reply:
> Commented code uses only `//`

We generally don't want commented code committed to the repo, so having it warn in CI seems fine.